app/src/main/assets/extensions/wolvic_autowebxr/content.js

```diff
@@ -0,0 +1,142 @@
+    return result.singleNodeValue;
+}
+
+function clickImmersiveElement() {
+    // Check if the current URL has extra query parameters
+    parentElementXPath = undefined;
+    targetElementXPath = undefined;
+
+    let url = document.URL;
+    let params = new URLSearchParams(new URL(url).search);
+    for (let [key, value] of params) {
+        if (key === PARENT_ELEMENT_XPATH_PARAMETER)
+            parentElementXPath = value;
+        else if (key === TARGET_ELEMENT_XPATH_PARAMETER)
+            targetElementXPath = value;
+    }
+
+    // We need at least the target element to click
+    if (!targetElementXPath)
+        return;
+
+    logDebug('Preparing to open immersive WebXR; parentElementXPath: ' + parentElementXPath + ' ; targetElementXPath: ' + targetElementXPath);
+
+    // The parent element is typically an iframe and, if it comes from a different origin,
+    // we might not be able to access its contents directly.
+    // If parentElementXPath is null, we will use the root to find the target element.
+
+    var parent, parentDocument;
+    if (parentElementXPath) {
+        parent = getElementByXPath(document, parentElementXPath);
+        if (!parent) {
+            logDebug('Parent element not found, retrying');
+            setTimeout(clickImmersiveElement, 1000);
+            return;
+        }
+
+        try {
+            parentDocument = parent.contentDocument || parent.contentWindow.document;
+        } catch (e) {
+            logDebug('Parent iframe is from a different origin');
+            const iframeWindow = parent.contentWindow;
+
+            const targetElementMsg = {
+                action: TARGET_ELEMENT_MSG,
+                targetElementXPath: targetElementXPath
+            };
+
+            iframeWindow.postMessage(targetElementMsg, '*');
+
+            // The iframe might not be ready yet, so we set up a listener for the "iframe ready" message.
+            const handleIframeReady = (event) => {
+                if (event.source === iframeWindow && event.data === IFRAME_READY_MSG) {
+                    window.removeEventListener('message', handleIframeReady);
+                    iframeWindow.postMessage(targetElementMsg, '*');
+                }
+            };
+            window.addEventListener('message', handleIframeReady);
+
+            return;
+        }
+    } else {
+        parent = window;
+        parentDocument = document;
+    }
+
+    if (parentDocument.readyState !== 'complete') {
+        logDebug('Parent is still loading');
+        parent.addEventListener('load', function() {
+            logDebug('Parent has finished loading');
+            clickImmersiveElement();
+        });
+        return;
+    } else {
+        logDebug('Parent is loaded');
+    }
+
+    let targetElement = getElementByXPath(parentDocument, targetElementXPath);
+
+    if (targetElement) {
+        logDebug('Target element found, calling click()');
+        targetElement.click();
+        targetElement.click();
```

Comment:
Why two?

Reply:
There was a case where one `click()` didn't work. I think that I will remove this and test again different experiences, to check if it is still needed.